autoload/qf.vim

```diff
@@ -173,17 +173,13 @@ function! qf#FormatColumn(col)
 endfunction
 
 function! qf#FormatType(type)
-    if a:type =~? 'e'
-        return ' error'
-    elseif a:type =~? 'i'
-        return ' info'
-    elseif a:type =~? 'n'
-        return ' note'
-    elseif a:type =~? 'w'
-        return ' warning'
-    else
-        return ''
-    endif
+    let types = {
+                \ 'e': ' error',
+                \ 'i': ' info',
+                \ 'n': ' note',
+                \ 'w': ' warning'
+                \ }
+    return a:type != '' ? types[a:type] : ''
```

Comment:
I've not yet pulled this to give a proper look but just looking here on GitHub this ternary seems a little odd. Wouldn't such a scenario is typically handled with a`get()`? The dictionary could even be inlined, e.g.

```
return get({ 'e': ' error',
           \ 'i': ' info',
           \ 'n': ' note',
           \ 'w': ' warning'
           \ }, a:type, '')
```

Reply:
You are right. Here is another one:

    return { 'e': ' error',
           \ 'i': ' info',
           \ 'n': ' note',
           \ 'w': ' warning'
           \ }->get(a:type, '')

I love `get()`.

Reply:
FWIW, I'm shooting for "current Vim", here, so all the legacy code and verbiage is going to be removed: `:Doline`/`:Dofile`, installation instructions for 7.4/8.0, etc.

I'm considering making a last backward-compatible release to vim.org with all the latest bug fixes and more version guards before releasing this modernised version.

Reply:
Fair enough, I thought as much when I saw the remark about using method syntax in the initial comment. I don't use it personally which is why I missed it in my previous example, though I do think it is one of the true improvements that's been made to Vim script.

Reply:
I changed the body of the function.